graphqlapi/src/main/java/io/stargate/graphql/schema/cqlfirst/dml/fetchers/MutationFetcher.java

```diff
@@ -74,9 +77,37 @@ protected MutationFetcher(
     }
 
     // Execute as a single statement
-    return dataStore
-        .execute(query)
-        .thenApply(rs -> ImmutableMap.of("value", environment.getArgument("value")));
+    return dataStore.execute(query).thenApply(rs -> toMutationResult(rs, environment));
+  }
+
+  private Map<String, Object> toMutationResult(
+      ResultSet resultSet, DataFetchingEnvironment environment) {
+
+    List<Row> rows = resultSet.currentPageRows();
+    Object originalValue = environment.getArgument("value");
+
+    if (rows.isEmpty()) {
+      // if we have no rows means that we got no information back from query execution, return
+      // original value to the user and applied true to denote that query was accepted
+      // not matter if the underlying data is not changed
+      ImmutableMap.Builder<String, Object> resultBuilder = ImmutableMap.builder();
+      resultBuilder.put("value", originalValue);
+      resultBuilder.put("applied", true);
+      return resultBuilder.build();
```

Comment:
Nit: this could also be shortened as
```suggestion
      return ImmutableMap.of("value", originalValue, "applied", true);
```
But the builder version is slightly more readable. Feel free to keep whatever you prefer.

Reply:
Yea, let's shorten it, less code, it's also readable without the builder.